### PR TITLE
Fix median/percentile across tier-1 databases (S7, #55)

### LINKED ISSUE #55: S7: Register `median`, `percentile_cont`, `percentile_disc` as SQLite UDFs at connect time
## Context
SQLite 3.45 (verified locally) ships native `EXP`/`LN`/`POWER`/`json_extract`/`->>`/`NTILE`/window functions, so most KB calculations are fine. The genuine residuals are `MEDIAN`, `PERCENTILE_CONT(p) WITHIN GROUP (ORDER BY x)`, and `PERCENTILE_DISC(p)`. Registering these as Python UDFs at connect time eliminates ~10 KB entries that would otherwise fall back to description text.

## Scope
- Hook the SQLite connection-creation point in SLayer's SQL client (likely `slayer/engine/sql_client.py`).
- Register per-`Connection` (not globally):
  - `median(x)` — `con.create_aggregate('median', 1, _MedianAgg)`
  - `percentile_cont(p, x)` and `percentile_disc(p, x)` — 2-arg aggregates
- Naming lowercase to align with sqlglot; sqlglot can rewrite Postgres `PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY x)` to a SQLite-friendly call.

## Implementation notes
- `_MedianAgg`: buffer values, compute on `finalize`. Buffer is fine for benchmark-scale tables.
- Empty input → NULL.
- Postgres path unchanged.

## Acceptance criteria
- [ ] Integration test: query using `median(x)` and `percentile_cont(0.5, x) WITHIN GROUP (ORDER BY x)` against in-memory SQLite via SLayer's engine.
- [ ] Doc note in `docs/database-support.md`.

## Depends on
None.

## Blocks
B6 (soft).

### PR DESCRIPTION
**Pull request by ZmeiGorynych**
## Summary
- **SQLite**: register `median` / `percentile_cont` / `percentile_disc` as Python aggregate UDFs on every new SQLite connection via SQLAlchemy's `connect` event (`slayer/sql/sqlite_udfs.py`).
- **ClickHouse**: percentile now emits the parametric `quantile(p)(x)` syntax instead of `PERCENTILE_CONT … WITHIN GROUP …`, which ClickHouse rejects.
- **MySQL**: `median` and `percentile` now raise `NotImplementedError` at SQL-generation time with an actionable message (point at MariaDB or client-side computation). MySQL has no native function and no Python-UDF mechanism, and the available workarounds (`GROUP_CONCAT`/windowed CTEs) all have material downsides.
- **Generator refactor**: pull `percentile` out of the static `BUILTIN_AGGREGATION_FORMULAS` into a new dialect-aware `_build_percentile` method, mirroring the existing `_build_median`. Both now branch on dialect.
- Closes #55, plus the broader tier-1 gaps that turned up while investigating it.

### Before / after

| Engine | `median` before | `median` after | `percentile` before | `percentile` after |
|---|---|---|---|---|
| Postgres | works | works | works | works |
| DuckDB | works | works | works | works |
| SQLite | runtime error | works (UDF) | runtime error | works (UDF) |
| ClickHouse | works | works | runtime error | works (`quantile(p)(x)`) |
| MySQL | runtime error | clear `NotImplementedError` | runtime error | clear `NotImplementedError` |

## Test plan
- [x] `poetry run pytest` (978 unit tests pass)
- [x] `poetry run pytest tests/integration/test_integration.py -m integration` (SQLite, all green)
- [x] `poetry run pytest tests/integration/test_integration_duckdb.py -m integration` (DuckDB, all green — including new regression tests)
- [x] `poetry run ruff check slayer/ tests/` clean
- [x] New tests:
  - `tests/test_sqlite_udfs.py` — 20 UDF unit tests (median/percentile_cont/percentile_disc, edge cases, GROUP BY state isolation)
  - `tests/test_sql_generator.py::TestMedianPercentilePerDialect` — 13 per-dialect SQL emission tests (incl. MySQL `NotImplementedError`)
  - `tests/integration/test_integration.py` — 5 SQLite integration tests (median, p25/p75 quartiles, empty-result NULL, grouped median, connection-pool reuse)
  - `tests/integration/test_integration_duckdb.py::TestDuckDBMedianPercentile` — 3 regression tests for the static-formula refactor
- [ ] Manual ClickHouse smoke (no integration suite exists for ClickHouse — covered by generator unit tests; recommend extending `examples/clickhouse/verify.py` separately)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added median and percentile aggregations with dialect-aware SQL; SQLite provides local implementations and MySQL is explicitly unsupported.

* **Documentation**
  * New database support guide and updated aggregation examples with per-engine compatibility, caveats, and a migration checklist.

* **Tests**
  * Added unit and integration tests covering median/percentile behavior across SQLite, DuckDB, Postgres, and ClickHouse.

* **Examples**
  * Updated verification scripts and examples to validate median/percentile results.

### CODE REVIEW

File: examples/verify_common.py [hunk 1]
```diff
@@ -159,6 +160,60 @@ def run_common_checks():
...
+    Inlined here so verify.py has no third-party dependencies.
+    """
+    s = sorted(values)
+    n = len(s)
+    if n == 1:
+        return s[0]
+    rank = p * (n - 1)
+    lo = int(rank)
+    hi = min(lo + 1, n - 1)
+    return s[lo] + (rank - lo) * (s[hi] - s[lo])
```
[the remark this answers is not in the document]
Comment (ZmeiGorynych) [reply examples/verify_common.py:175]: Skipping this one. `_percentile_cont` is a private helper inside `examples/verify_common.py`; its only caller passes `values = [o[3] for o in ORDERS]` (non-empty by construction, since the seed always populates `ORDERS`) and `p` is always a hard-coded `0.25` / `0.75` literal. Per the repo's coding conventions we avoid adding validation for inputs that cannot occur — this helper isn't a public API, so empty `values` or out-of-range `p` cannot reach it without an unrelated bug elsewhere. Happy to revisit if `_percentile_cont` ever grows external callers.
